### DISCUSSION TITLE
Nintendo Switch as a supported console via TinyUSB

### DISCUSSION THREAD

**Discussion by coolbho3k** [darthcloud/BlueRetro#665] [Ideas]
I got this idea from this repo: https://github.com/DavidPagels/retro-pico-switch which adapts an N64 or GameCube controller to a USB Switch controller.

It's apparently possible to act as a USB controller for the Nintendo Switch via TinyUSB, which is available for ESP32.

Unlike a lot of modern consoles, I don't believe the Nintendo Switch has any DRM on USB controllers.

This would make BlueRetro an open source alternative for the 8bitdo USB Bluetooth receiver, and support all the controllers that BlueRetro supports.

One challenge I foresee: BlueRetro does not support gyro axis yet and getting it to work properly might be annoying. Translation from HD Rumble signals the Switch sends to, for example, DualSense (or even Switch Pro Controller) may or may not be possible. Another challenge: probably only Switch controller per ESP32 is supported.

I know Switch is not "retro" yet (but it will be one day!). Even today, this would be pretty neat and probably useful for some.

If this is a good idea, I can begin looking into this.

**Comment by darthcloud** [comment]
The original ESP32 do not have an USB PHY, and all other ESP lack BT classic support.

Your best bet is to do the usb on a 2nd mcu like a pico and talk to BlueRetro via SPI

**Comment by coolbho3k** [comment]
Ah crap, I didn't realize that only S2 and S3 have it. 

SPI output to another MCU would be a cool project.
